### PR TITLE
Azure File Share Healthcheck

### PR DESCRIPTION
**Pull request by JonaPou**
**What this PR does / why we need it**:
This pull request adds a health check for Azure file shares using the "exists" action from the SDK.

**Which issue(s) this PR fixes**:
None.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**: 
No.

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [ ] Created/updated tests
- [x] Unit tests passing
- [ ] End-to-end tests passing
- [x] Extended the documentation
- [ ] Provided sample for the feature

### CODE REVIEW

File: src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var shareClient = new ShareClient(_connectionString, _shareName);
+
+                if (!await shareClient.ExistsAsync(cancellationToken))
+                {
+                    return new HealthCheckResult(context.Registration.FailureStatus, description: $"File Share '{_shareName}' does not exists");
```
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs:28]:
```suggestion
                    return new HealthCheckResult(context.Registration.FailureStatus, description: $"File Share '{_shareName}' does not exist");
```

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs [hunk 2]
```diff
@@ -63,6 +64,29 @@ public static IHealthChecksBuilder AddAzureBlobStorage(this IHealthChecksBuilder
                timeout));
         }
 
+        /// <summary>
+        /// Add a health check for an Azure file share.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <param name="connectionString">The Azure Storage connection string to be used.</param>
+        /// <param name="shareName">The name of the Azure file share to check if exist.</param>
+        /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azureblob' will be used for the name.</param>
```
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs:73]:
```suggestion
        /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azurefileshare' will be used for the name.</param>

```

File: src/HealthChecks.AzureStorage/HealthChecks.AzureStorage.csproj [hunk 3]
```diff
@@ -8,7 +8,8 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Azure.Storage.Blobs" Version="$(AzureStorageBlobs)" />
-    <PackageReference Include="Azure.Storage.Queues" Version="$(AzureStorageQueues)"/>
+    <PackageReference Include="Azure.Storage.Files.Shares" Version="12.8.0" />
```
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/HealthChecks.AzureStorage.csproj:11]: Please move this version inside props file to keep current design.

File: src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
+            _shareName = shareName;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var shareClient = new ShareClient(_connectionString, _shareName);
```
Comment (sungam3r) [inline src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs:24]: Are you sure you want to create an instance of `ShareClient` each healthcheck request?

File: src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs (hunk 1 above)
Comment (JonaPou) [reply src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs]: Fixed it

File: src/HealthChecks.AzureStorage/HealthChecks.AzureStorage.csproj (hunk 3 above)
Comment (JonaPou) [reply src/HealthChecks.AzureStorage/HealthChecks.AzureStorage.csproj]: Fixed it

File: src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs (hunk 4 above)
Comment (JonaPou) [reply src/HealthChecks.AzureStorage/AzureFileShareHealthCheck.cs]: Fixed it

File: src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs (hunk 2 above)
Comment (JonaPou) [reply src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs]: Fixed it

**Review by sungam3r — APPROVED** [review]

**Comment by sungam3r** [comment]
@unaizorrilla Should all/new PRs be targeted against `netcore-6.0` branch ?

**Comment by unaizorrilla** [comment]
We need:

 - Merge on master ( currently 5.0 ) and create a tag for release it with appveyord
 - Pull down to net core 6 (set version correctly)

**Comment by sungam3r** [comment]
I would like to look from the side.

**Comment by carlosrecuero** [comment]
Hi @JonaPou , your PR has been merged and the related nuget package published with version 6.0.3. Thank you for your contribution.

Thank you for handling this PR, @sungam3r .

**Pull request merged**